processing/src/main/java/io/druid/query/aggregation/hyperloglog/HyperUniquesAggregator.java

```diff
@@ -57,7 +57,8 @@ public void reset()
   @Override
   public Object get()
   {
-    return collector;
+    // Workaround for OnheapIncrementalIndex's penchant for calling "aggregate" and "get" simultaneously.
+    return HyperLogLogCollector.makeCollector(collector.getStorageBuffer().duplicate());
```

Comment:
HLL is already super expensive to calculate, can you get some benchmarks on how this effects both ingestion and querying?